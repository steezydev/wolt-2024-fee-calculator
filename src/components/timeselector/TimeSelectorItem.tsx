import ButtonUnstyled from '@/components/button/ButtonUnstyled';
import { classNames } from '@/helpers/classNames';
import { TimeSelectorItemProps } from '@/types/props/TimeSelectorItemProps';

const TimeSelectorItem = ({
  value,
  type,
  isActive,
  onClick,
}: TimeSelectorItemProps) => (
  <ButtonUnstyled
    id={`select${type.charAt(0).toUpperCase() + type.slice(1)}${value}Button`}
    ariaLabel={`Select ${value} ${type}`}
    type='button'
    className={classNames(
      'px-3 py-1 rounded-lg hover:bg-primary-100 w-full text-black dark:text-white text-base',
      isActive && 'bg-primary-300 text-white hover:bg-primary-300'
    )}
    onClick={onClick}
  >
    {value.toString().padStart(2, '0')}
  </ButtonUnstyled>
);

export default TimeSelectorItem;
