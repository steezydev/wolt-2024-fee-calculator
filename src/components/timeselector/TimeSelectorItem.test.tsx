import {
  timeSelectorItemEventTests,
  timeSelectorItemRenderTests,
} from './TimeSelector.tests';
import TimeSelectorItem from './TimeSelectorItem';

timeSelectorItemRenderTests('TimeSelectorItem', TimeSelectorItem);
timeSelectorItemEventTests('TimeSelectorItem', TimeSelectorItem);
