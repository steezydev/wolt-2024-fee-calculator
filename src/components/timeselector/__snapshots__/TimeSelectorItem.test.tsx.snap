// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSelectorItem when asked to render type hour active renders type hour active 1`] = `
<button
  aria-label="Select 15 hour"
  className="px-3 py-1 rounded-lg hover:bg-primary-100 w-full text-black dark:text-white text-base bg-primary-300 text-white hover:bg-primary-300"
  data-test-id="selectHour15Button"
  id="selectHour15Button"
  onClick={[Function]}
  type="button"
>
  15
</button>
`;

exports[`TimeSelectorItem when asked to render type hour renders type hour 1`] = `
<button
  aria-label="Select 15 hour"
  className="px-3 py-1 rounded-lg hover:bg-primary-100 w-full text-black dark:text-white text-base"
  data-test-id="selectHour15Button"
  id="selectHour15Button"
  onClick={[Function]}
  type="button"
>
  15
</button>
`;

exports[`TimeSelectorItem when asked to render type minute active renders type minute active 1`] = `
<button
  aria-label="Select 35 minute"
  className="px-3 py-1 rounded-lg hover:bg-primary-100 w-full text-black dark:text-white text-base bg-primary-300 text-white hover:bg-primary-300"
  data-test-id="selectMinute35Button"
  id="selectMinute35Button"
  onClick={[Function]}
  type="button"
>
  35
</button>
`;

exports[`TimeSelectorItem when asked to render type minute renders type minute 1`] = `
<button
  aria-label="Select 35 minute"
  className="px-3 py-1 rounded-lg hover:bg-primary-100 w-full text-black dark:text-white text-base"
  data-test-id="selectMinute35Button"
  id="selectMinute35Button"
  onClick={[Function]}
  type="button"
>
  35
</button>
`;
